### PR TITLE
stm32: remove zephyr specific IRQ defines

### PR DESCRIPTION
**Pull request by erwango**
Remove zephyr specific IRQ definitions and replace with CMSIS ones.
Even if these should be removed eventually by device tree generated constants, it will save some effort when introducing new STM32 series (F7, F2 atm)

### CODE REVIEW

**Comment by tagunil** [comment]
@erwango, there are no CMSIS definitions for STM32F4 EXTI16 to EXTI22. Why should these internally connected lines' handlers be enabled here in the first place?

**Comment by zhaozhongchen** [comment]
@tagunil @erwango EXTI16 to EXTI22 are shared with others irq on F4 and F2, there should has a define.

**Comment by galak** [comment]
How about adding DT support for the DMA instead, and getting IRQs from there.

File: drivers/interrupt_controller/exti_stm32.c [hunk 1]
```diff
@@ -113,56 +113,56 @@ void stm32_exti_enable(int line)
 	}
 #elif CONFIG_SOC_SERIES_STM32F1X
```
Comment (zhaozhongchen) [inline drivers/interrupt_controller/exti_stm32.c]: How about merge EXTI15..0 for F1, F3, F4? eg:
```
#if CONFIG_SOC_SERIES_STM32F1X || CONFIG_SOC_SERIES_STM32F3X || CONFIG_SOC_SERIES_STM32F4X
if (line >= 5 && line <= 9) {
    irqnum = EXTI9_5_IRQn;
} else if 
```

**Comment by erwango** [comment]
> How about adding DT support for the DMA instead, and getting IRQs from there.

@galak , this would require to move DMA to Cube LL first (to ensure we get correct DT bindings). I don't know when it will be done as it's not the first prio. Besides this should be done for exti as well.
Meantime people keep adding STM32XX_IRQ when pushing new series (F2, F7 on going right now)...

Would be nice to get rid of these, since work is ready. Holding it will not give me time to work on DMA, just more work in reviews to know if people manually coded the IRQ right.

File: drivers/interrupt_controller/exti_stm32.c (hunk 1 above)
Comment (erwango) [reply drivers/interrupt_controller/exti_stm32.c]: Let's get it in tree, then we'll be able to proceed with optimisations.

**Comment by tagunil** [comment]
@erwango, I don't understand how F4 is different from e.g. L0 regarding EXTI >= 16 mapping. Why we enable these IRQs for F4, but not for other families?

**Comment by erwango** [comment]
@tagunil , well I think there is no good reason and that has to be reviewed. Which should be easier once this PR is merged

**Review by galak — APPROVED** [review]

**Pull request merged**